### PR TITLE
Stop early swipes causes crashes

### LINKED ISSUE #4896: Prevent swipes before form has loaded properly
It seems like from this [crash](https://console.firebase.google.com/u/0/project/api-project-322300403941/crashlytics/app/android:org.odk.collect.android/issues/30a3178c00619de299e65f60078556f8?time=last-seven-days&sessionEventKey=6184F4B302540001171121663967053E_1605465151459213822) that it's possible to swipe before the form has loaded (and there's no view on screen). This causes a `NullPointerException` as `currentView`  (in `FormEntryActivity`) hasn't been assigned yet.

It's hard to reproduce this, but it should be easy enough to protect against, and it does seem like a nasty instability in our swipe code.

## Notes

I think fixing this is a nice opportunity to encapsulate our swipe code (scattered currently between `SwipeHandler` and various methods in `FormEntryActivity`) into a dedicated view, separating swiping from the actual view creation and form index manipulation entirely. In that world it'd be easy to just contain `currentView` in there and ignore swipes or next/back clicks if the view hasn't been populated. In my mind, a possibility would be something like this:

```kotlin
class SwipeView {

    // For refreshing, setting first view
    fun setView(view: View)

    // For explicit moves forward and backwards like when prompting for a repeat.
    // This might get rid of the need for `FormIndexAnimationHandler`
    fun setView(view: View, direction: Direction)

    // Called by nav bar buttons. These would then trigger the appropriate listener method
    fun next()
    fun back()
   
    // The `SwipeView` can use the listener to "ask" for the next/previous view. 
    // `FormEntryActivity` could implement the methods and carry out the form index
    // manipulation, create the view and then return it in each.
    fun setListener(listener: Listener) 

    interface Listener {

        fun onSwipeForward(): View

        fun onSwipeBackward(): View
    }
}
```

We've chatted about moving to a `ViewPager` before, but that process seems to be so hard at this point that I think doubling down on a custom swipe system but extracting it as a reusable component (probably in its own module) would at least give us some separation and could untangle some of `FormEntryActivity`.

### PR DESCRIPTION
**Pull request by seadowg**
Closes #4896

#### What has been done to verify that this works as intended?

Verified manually (with modified code to create a window of time when you can swipe before the form has loaded).

#### Why is this the best possible solution? Were any other approaches considered?

The issue lays out a nice (if I do say so myself), but more time intensive route to solving this. We made some good progress on cleaning up the swipe code in the last release however, and I think we should see how further loading states (like we're adding in #5229) force us to keep making progress there.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Should have no effect although changing form entry swipe code is always a little scary. Good to put form entry and swiping/using next and back buttons through its paces.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java [hunk 1]
```diff
@@ -1492,7 +1492,7 @@ public void showView(SwipeHandler.View next, AnimationType from) {
         // adjust which view is in the layout container...
         SwipeHandler.View staleView = currentView;
         currentView = next;
-        swipeHandler.setView(currentView);
+//        swipeHandler.setView(currentView);
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java]: This should be reverted I guess.

Comment (seadowg) [reply collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java]: Ach I forgot to redo `git add` after taking this out again.

**Review by grzesiek2010 — APPROVED** [review]

**Comment by grzesiek2010** [comment]
I think it doesn't need testing so I'm ok with merging but if you want the QA team to take a look at it please add `needs testing` label.

**Comment by seadowg** [comment]
> I think it doesn't need testing so I'm ok with merging but if you want the QA team to take a look at it please add needs testing label.

I mostly agree with you, but I'm feeling nervous about it after my last set of changes to `FormEntryActivity` so going to let QA have a look.

**Comment by dbemke** [comment]
**Tested with success!**

Verified on Androids: 8.1 and 11.

**Tested scenarios:**

- try to reproduce #4896 issue 
- swipe forward/backward 
- next/back button
- try with different forms 
- big forms
- Form Entry Settings: Check/Uncheck Moving backwards button
- Form Management: different Constraint processing behavior

**Comment by srujner** [comment]
**Tested with success!**

Verified on Androids: 5.1, 10, 12

**Pull request merged**